### PR TITLE
chore: normalize test config

### PR DESCRIPTION
**Pull request by boneskull**
This change does the following:

1. Upgrade `ava`.
2. Rename all test file extensions to `.spec.js`.
3. Relocate all Ava config into the `package.json` of each workspace.
4. Change some commented-out tests to skipped tests.
5. BONUS: Config to run all tests via e.g., `npm exec ava` from the workspace root

### CODE REVIEW

**Comment by boneskull** [comment]
Current dependencies on/for this PR:
* main
  * **PR #629** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/629" target="_blank">Graphite</a> 
    * **PR #653** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/653" target="_blank">Graphite</a> 
    * **PR #643** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/643" target="_blank">Graphite</a>  👈
      * **PR #620** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/620" target="_blank">Graphite</a> 
    * **PR #632** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/632" target="_blank">Graphite</a> 
    * **PR #631** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/631" target="_blank">Graphite</a> 
    * **PR #630** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/630" target="_blank">Graphite</a> 
    * **PR #627** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/627" target="_blank">Graphite</a> 
      * **PR #639** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/639" target="_blank">Graphite</a> 
      * **PR #617** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/617" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/643?utm_source=stack-comment).

File: packages/aa/package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   },
   "scripts": {
     "test": "npm run test:run",
-    "test:run": "ava --timeout=30s test/index.js",
+    "test:run": "ava",
```
Comment (naugtur) [inline packages/aa/package.json:14]: (can be ignored) IMHO the arguments are more convenient as one can copy the call and modify/extend without thinking of the configuration elsewhere, but I'm ok with this change as it provides benefits elsewhere. Consistency is more valuable than copy-pasting.

File: packages/lavapack/test/runtimeEcma.spec.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const espree = require('espree')
+
+const ECMA_VERSION = 2020;
+
+test('runtime.js parses as ECMA 2020', (t) => {
+  const code = fs.readFileSync(require.resolve('../src/runtime.js'), 'utf8')
+  t.notThrows(() => {
+    espree.parse(code, {
+    ecmaVersion: ECMA_VERSION,
+    })
```
Comment (naugtur) [inline packages/lavapack/test/runtimeEcma.spec.js:14]: (should) This test used to be more informative. 
It's likely that if this test fails, it does so unexpectedly and a more descriptive message (ie containing the runtime.js` filename) is going to be helpful for the person encountering it.

File: packages/tofu/test/util.spec.js [hunk 3]
```diff
@@ -1,7 +1,15 @@
 const test = require('ava')
 const { utils: { mergePolicy, mergeConfigEntire, objToMap, mapToObj } } = require('../src/index')
 
-testMergePartial('upgrades reads to writes', {
+const testMergePartial = test.macro(
```
Comment (naugtur) [inline packages/tofu/test/util.spec.js:4]: Nice! when was it introduced? I haven't seen test.macro used before.

**Review by naugtur — APPROVED** [review]

File: packages/lavapack/test/runtimeEcma.spec.js (hunk 2 above)
Comment (boneskull) [reply packages/lavapack/test/runtimeEcma.spec.js:14]: - [x] I'll restore the custom failure output, but keep it in an ava test.

File: packages/tofu/test/util.spec.js (hunk 3 above)
Comment (boneskull) [reply packages/tofu/test/util.spec.js:4]: I have no idea, I'm new to `ava`, but discovered this while I was poking around.

**Pull request merged**